index.js

```diff
@@ -1,14 +1,23 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
-		get: function (target, name, receiver) {
+module.exports = function (target) {
+	var handler = {
+		get: function (target, name) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
-			return Reflect.get(target, name, receiver);
+			return Reflect.get(target, name);
 		}
-	});
+	};
+
+	var instance = new Proxy(target, handler);
+
+	instance.decorateWith = function (fn) {
```

Comment:
Maybe I'm missing the point here, but the difficulties are a bit connected to this [discussion](https://github.com/wycats/javascript-decorators/issues/7).

If i try to do something like this and export it 
```javascript 
function isTestable(value) {
   return function decorator(target) {
      target.isTestable = value;
   }
}
```
i'll end up with an.. say ```export.decorate = function...etc``` and the calling of known (correct me if i'm wrong) will prevail with ```module.export``` or if i turn the export into an object and expose both methods the invoking will not be the same.

If these concerns are not correct or not applicable i'll drop the PR and maybe someone can do the proper implementation requested of this